src/pages/Dashboard.tsx

```diff
@@ -122,6 +172,38 @@ const Dashboard = () => {
             <Button onClick={createNewInsight}>
                 New Insight
             </Button>
+            {/**TODO: Popup form for new group**/}
+            <div>
+                <form onSubmit={() => createNewGroup(newGroupNameInput)}>
+                    <label>
+                        New Group Name:
+                        <input
+                            type="text"
+                            value={newGroupNameInput}
+                            onChange={changeEvent => setNewGroupNameInput(changeEvent.target.value)}
+                            minLength={3}
+                            maxLength={25}
+                        />
+                    </label>
+                    <input type="submit" value="Submit" />
+                </form>
+            </div>
+            <div>
+                <span>Groups:</span>
+                <ul>
+                    {currentUserGroups.map(group => (
+                        <li
+                            style={currentViewedGroup === group ? {color: 'blue'} : {cursor: 'pointer'}}
+                            onClick={currentViewedGroup !== group ? async () => {await switchViewedGroup(group);} : undefined}
```

Comment:
Technically I don't think you need the ternary here if you move `loadRequestsAndInsights` into a useEffect, you are still calling a state update but since you're passing the same string as the current state value React will detect that and not force any rerenders